### PR TITLE
chore: Moves Fade component into TableElement

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Moves `Fade` component into `TableElement`. It didn't make the cut to be a component in itself 😆.

### TEST PLAN
1 - Execute `TableElement` tests.
2 - All tests should pass.

@rusackas @junlincc 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
@yardz @geido let's review. 🙏

**Review by geido — APPROVED** [review]
I have mixed feelings about this. I know the component is negligible in size and purpose but I think I like the fact that you can use it as a small utility component. Either way, the code LGTM and it can go if it reaches consensus.

**Comment by michael-s-molina** [comment]
> I have mixed feelings about this. I know the component is negligible in size and purpose but I think I like the fact that you can use it as a small utility component. Either way, the code LGTM and it can go if it reaches consensus.

@geido I would say that when you look at the component's name, you can imagine a reusable component used in multiple places. But when you look at the implementation and props you will see that is just a div with almost no behavior and used only in `TableElement`. I have difficulty imagining creating tests and a storybook for this at its current stage. Maybe in the future, if a Fade component is required in other places and with more behavior, we can move it to `src/components`.

**Comment by yardz** [comment]
> @geido I would say that when you look at the component's name, you can imagine a reusable component used in multiple places. But when you look at the implementation and props you will see that is just a div with almost no behavior and used only in `TableElement`. I have difficulty imagining creating tests and a storybook for this at its current stage. Maybe in the future, if a `Fade` component is required in other places and with more behavior, we can move it to `src/components`.

I have a very strong opinion about not assembling components (to be honest i will always be against it).
I did a very quick and superficial search and saw several components using "some kind" of fade... Which brings me to the question, why this fade is not being used in more places?
I believe that a piece of the reason is precisely because there is no fade in `src/components ` and the idea that "if a component is used only here, I will put it here".

But I don't see anything wrong with the code. I just disagree with the PR's goal.

**Comment by michael-s-molina** [comment]
> I have a very strong opinion about not assembling components (to be honest i will always be against it).

For me, always is a very strong word in the context of software development. I tend to analyze case by case considering the context.

> I did a very quick and superficial search and saw several components using "some kind" of fade... Which brings me to the question, why this fade is not being used in more places?

Thank you for the search. That's a strong indicator that in fact, we should have a `Fade` component in `src/components`. Can you paste the references here so we can close this PR and open a new one with the objective of creating this new Fade component and consolidating all the references?

> I believe that a piece of the reason is precisely because there is no fade in `src/components ` and the idea that "if a component is used only here, I will put it here".

I understand that if we already have a component available that will prevent duplications, but this does not mean that every little bit of code that we do should be in the components folder. Generally, when we have classical components like Button we start there, but a lot of components evolve to a shared component and end up in the components folder by merit. I think that this is the case with Fade. If we identify multiple places with this behavior and resolve all use cases using a nice API, then we can move there.

**Comment by yardz** [comment]
> Thank you for the search. That's a strong indicator that in fact, we should have a `Fade` component in `src/components`. Can you paste the references here so we can close this PR and open a new one with the objective of creating this new Fade component and consolidating all the references?

These components have some "fade behavior":
- `src/dashboard/components/gridComponents/ChartHolder.jsx`
- `src/chart/Chart.jsx`
- `src/messageToasts/components/ToastPresenter.tsx`

There are also "fade effects" in less files that I was unable to find the components that use them. Again, this search was very superficial (I searched for the name "fade" And I did a very simple reading of the code). If we look for "transition" + "opacity" effects, I believe we will find more.

**Comment by yardz** [comment]
> For me, always is a very strong word in the context of software development. I tend to analyze case by case considering the context.

I agree 100%, that's why I said I have a very strong opinion in this case hahaha
In this case I really think it’s **always**

But again, this is my **opinion** and not a **pattern of this project**, you are totally free to disagree and this is not a block for the merge.

**Review by rusackas — APPROVED** [review]
I think the way this fade component is used, with all of the state management and so forth, is probably more complex than it needs to be. We can circle back later, but I'm willing to bet a refactor of this could greatly simplify things.

TL;DR: LGTM.
